### PR TITLE
[Android] ClearButtonVisibility with margin fix

### LINKED ISSUE #25225: In the component Entry, ClearButtonVisibility="WhileEditing" doesn't work.
### Description

Using `Entry` component, with the property `ClearButtonVisibility="WhileEditing"` doesn't work.

Context : 
VS 2022 17.11.5
Net 8.0
NuGet Microsoft.Maui.Controls 8.0.91
Emulator Pixel 5 Android 14 API 34

### Steps to Reproduce

Create a new MAUI default project in .Net 8.
In the MainPage.xaml, add a component `Entry `and add the property `ClearButtonVisibility="WhileEditing"`.
`<Entry Text="text to clear" ClearButtonVisibility="WhileEditing"/>`

Write some text an try to click on the cross/clear button : Nothing happens.

### Link to public reproduction project repository

_No response_

### Version with bug

Unknown/Other

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

_No response_

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

No

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/25225

|Before|After|
|--|--|
|<video src="https://github.com/user-attachments/assets/fe285e90-ac5d-4ddf-bbd4-fe2361574735" width="300px"/>|<video src="https://github.com/user-attachments/assets/a5b31b9f-1743-43ed-aaeb-fe0f451a2778" width="300px"/>|

### CODE REVIEW

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue17453.cs [hunk 1]
```diff
@@ -64,6 +64,33 @@ public void EntryClearButtonWorks()
...
+			// Set focus
+			App.TapCoordinates(entryRect.X, entryRect.Y);
+
+			// Tap Clear Button
+			var margin = 30;
+			App.TapCoordinates(entryRect.X + margin, entryRect.Y + margin);
+
+			entryText = App.FindElement("EntryWithMargin").GetText();
+
+			ClassicAssert.IsEmpty(entryText);
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue17453.cs:92]: The test is failing on Android API 30
image

Comment (kubaflo) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue17453.cs:91]: @jsuarezruiz can you please /azp?

**Review by jsuarezruiz — APPROVED** [review]

**Review by jfversluis — APPROVED** [review]

**Pull request merged**